src/Microsoft.ML.SearchSpace/Schema/SchemaBase.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.ML.SearchSpace.Schema
+{
+    public abstract class SchemaBase
```

Comment:
`SchemaBase` doesn't seem like a great name.

1. It is a pretty general term, and doesn't really convey what it is.
2. It is confusing between this and `DataViewSchema`. We renamed `DataViewSchema` from just `Schema` in order to give it context, since "Schema" is such a common/general term.

Reply:
Will `SearchSpaceSchemaBase` be better here?

Reply:
Can you describe what this class is supposed to represent?

With a name like `SearchSpaceSchema`, it might not need `Base` in the name.

Reply:
It seems like you are putting the version both in the class name and as a member in the class. Do you need both?

Reply:
> Do you need both?

Yeah I think so. The version suffix in class name is more of a naming style, it's not used anywhere

Reply:
The class is internal right? Are you planning on making a new schema class every time you have a new schema?

Reply:
> The class is internal right? Are you planning on making a new schema class every time you have a new schema?

Yep and yep, the old schema needs to be kept so we can load from files using that schema.